### ISSUE TITLE
Introducing Layout XML-defined cells into legacy TableView

### ISSUE THREAD

**Issue by tomekc** [schibsted/layout#74]
I am very impressed by the work and ingenuity behind Layout. So much I decided to try introducing it into one of projects.

Unfortunately, I can't get UITableViewCell laid out properly: the content of XML is loaded, but  everything is crammed in top-left corner. I started learning by modifying example app and tinkering with TableCell.xml, and got that:

<img width="426" alt="screen shot 2017-10-31 at 13 47 20" src="https://user-images.githubusercontent.com/407463/32224651-a82e7d8a-be42-11e7-8c0b-1dade32cf310.png">

Then, using the same XML in my legacy project (existing UITableView, existing reusable cells defined in storyboard) result with this:

<img width="407" alt="screen shot 2017-10-31 at 13 36 09" src="https://user-images.githubusercontent.com/407463/32224684-c1089002-be42-11e7-919d-d058eac260af.png">

There's ScrollView wrapping UIDatePicker, but it has zero size in my project.

I do register cell ("node"):
```swift
@IBOutlet var tview: UITableView! {
        didSet {
            tview.registerLayout(named: "DelayedTimePickerCell.xml", forCellReuseIdentifier: CellIdentifiers.TimePicker)
        }
    }
```

and dequeue in cellForRow: 

```swift
let node = tableView.dequeueReusableCellNode(withIdentifier: CellIdentifiers.TimePicker, for: indexPath)
                node.setState([])
                return node.view as! UITableViewCell
```

The only part I am missing is not using LayoutViewController - do I have to, even if I only wanted to use cells? I will appreciate any pointers to what I am doing wrong.

**Comment by nicklockwood** [comment]
It would be useful if you could include the XML for your cell, but my instinct is that you may need to set the width and height of the cell (probably to 100% and auto, respectively).

**Comment by tomekc** [comment]
Here's the XML. It is a result of tinkering with TableCell.xml in sample app (and renders fine there):

```xml
<UITableViewCell
    height="auto + 20"
    reuseIdentifier="DelayedTimePicker"
    selectedBackgroundView.backgroundColor="colors.red"
    style="default"
    textLabel.highlightedTextColor="white"
    >
    
    <UISegmentedControl items="'Preset', 'Custom'"
    top="15"
    left="50% - (width / 2)"
    />
    
    <UIScrollView
        top="previous.bottom + 10"
        height="216"
        width="100%"
        contentSize.width="200%"
        isPagingEnabled="true"
        showsHorizontalScrollIndicator="false"
        >
        
        <UIDatePicker
        width="100%"
        height="auto"
        left="0"
        top="0"
        />
        
        <UIView width="100%"
            left="previous.right + 0"
            height="auto"
            backgroundColor="#FFE1FE"
            />
    </UIScrollView>
    
</UITableViewCell>
```

**Comment by nicklockwood** [comment]
I've tried recreating your cell in a new project with an ordinary UITableView created programmatically (not Layout-based) and it seems fine.

I guess it may be some sort of interaction with Storyboards? I'll keep looking.

**Comment by nicklockwood** [comment]
Hmm, it's also working when I use a storyboard.

Can you zip up your project and attach it? I don't think I can reproduce the issue myself.

**Comment by tomekc** [comment]
Legally I cannot share my project, but I while trying to pinpoint the root cause I managed to create a fresh project, and still the issue persists.

You can find it here: https://github.com/tomekc/LayoutProblemDemo

**Comment by nicklockwood** [comment]
Perfect, thanks 👍

**Comment by tomekc** [comment]
I think I am close to a solution. 

* I wrapped contents of my cell into `UIView`
* The wrapper UIView needs explicitly specified height, if I set `auto`, the app is crashing at assertion in *LayoutNode.swift* line 751 (`assert(_evaluating.isEmpty)`)

I guess that `UIDatePIcker` inside `ScrollView` below `SegmentedControl` is not the trivial use case :)

```xml
<UITableViewCell
    height="auto + 20"
    width="auto"
    reuseIdentifier="DelayedTimePicker"
    style="default"
    textLabel.highlightedTextColor="white"
    >
    
    <UIView left="0" width="100%" top="0" height="250" backgroundColor="#fff" >
        
        <UISegmentedControl
        items="'Custom', 'Preset'"
        top="10"
        left="50% - (width/2)"
        />
        
        <UIScrollView
            top="previous.bottom + 10"
            height="216"
            width="100%"
            contentSize.width="200%"
            isPagingEnabled="true"
            showsHorizontalScrollIndicator="false"
            >
            
            <UIDatePicker
            width="100%"
            height="auto"
            left="0"
            top="0"
            />
            <UIView width="100%"
            left="previous.right + 0"
            height="auto"
            backgroundColor="#FFE1FE"
            />
        </UIScrollView>               
    </UIView>   
</UITableViewCell>


```

**Comment by nicklockwood** [comment]
@tomekc the only bug I found in your original example was this line:

        selectedBackgroundView.backgroundColor="colors.red"

Since you don't define the "colors.red" constant anywhere, it was throwing an error. But because the project doesn't use LayoutViewController, the error was being swallowed silently.

Removing that line fixed the layout for me. There was no need to make any other changes.

I'll think about whether there is a better way to surface errors in projects without a LayoutViewController, but in the meantime you can set a breakpoint for Swift Error and it will help you to catch such issues.

**Comment by nicklockwood** [comment]
@tomekc sorry, you were right about the crash also. I was testing on my local branch, which has already fixed that.

If you link against the develop branch by putting this in your Podfile, you can use the fixed version:

    pod 'Layout', :git => 'https://github.com/schibsted/Layout.git', :branch => 'develop'

**Comment by nicklockwood** [comment]
Fixed in 0.5.9
